sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -1126,6 +1127,57 @@ public Query apply(Function parent, Function inner, Context context) {
             }
         }
 
+        static class SubscriptQuery implements InnerFunctionToQuery {
+
+            private static final ImmutableMap<String, BooleanClause.Occur> FUNCTION_TO_CLAUSE_OCCURRENCE =
+                ImmutableMap.<String, BooleanClause.Occur>builder()
+                    .put(EqOperator.NAME, BooleanClause.Occur.MUST)
+                    .put(GteOperator.NAME, BooleanClause.Occur.MUST)
+                    .put(GtOperator.NAME, BooleanClause.Occur.MUST_NOT)
+                    .put(LteOperator.NAME, BooleanClause.Occur.MUST)
+                    .put(LtOperator.NAME, BooleanClause.Occur.MUST_NOT)
+                    .build();
+
+            @Nullable
+            @Override
+            public Query apply(Function parent, Function inner, Context context) throws IOException {
+                assert inner.info().ident().name().equals(SubscriptFunction.NAME) :
+                    "function must be " + SubscriptFunction.NAME;
+
+                RefLiteralPair innerPair = new RefLiteralPair(inner);
+                if (!innerPair.isValid()) {
+                    return null;
+                }
+
+                Reference reference = innerPair.reference();
+                if (DataTypes.isCollectionType(innerPair.reference().valueType())) {
+                    BooleanClause.Occur clauseOccurrence = FUNCTION_TO_CLAUSE_OCCURRENCE.get(parent.info().ident().name());
+                    if (clauseOccurrence == null) {
+                        return null;
+                    }
+
+                    FunctionLiteralPair functionLiteralPair = new FunctionLiteralPair(parent);
+                    if (!functionLiteralPair.isValid()) {
+                        return null;
+                    }
+                    Input input = functionLiteralPair.input();
+
+                    MappedFieldType fieldType = context.getFieldTypeOrNull(reference.column().fqn());
+                    if (fieldType == null) {
+                        if (CollectionType.unnest(reference.valueType()).equals(DataTypes.OBJECT)) {
+                            return null; // fallback to generic query to enable {x=10} = any(objects)
```

Comment:
Should we have a test for this case? And the example in the comment seems wrong?